packages/accounts-2fa/2fa-server.js

```diff
@@ -0,0 +1,132 @@
+import twofactor from 'node-2fa';
+import QRCode from 'qrcode-svg';
+import { Meteor } from 'meteor/meteor';
+
+Accounts._is2faEnabledForUser = selector => {
+  if (!Meteor.isServer) {
+    throw new Meteor.Error(
+      400,
+      'The function _is2faEnabledForUser can only be called on the server'
+    );
+  }
+
+  if (typeof selector === 'string') {
+    if (!selector.includes('@')) {
+      selector = { username: selector };
+    } else {
+      selector = { email: selector };
+    }
+  }
+
+  const user = Meteor.users.findOne(selector) || {};
+  const { services: { twoFactorAuthentication } = {} } = user;
+  return (
+    twoFactorAuthentication &&
+    twoFactorAuthentication.secret &&
+    twoFactorAuthentication.type === 'otp'
+  );
+};
+
+Accounts._generate2faToken = secret => twofactor.generateToken(secret);
+
+Accounts._isTokenValid = (secret, code) => {
+  if (!Meteor.isServer) {
+    throw new Meteor.Error(
+      400,
+      'The function _isTokenValid can only be called on the server'
+    );
+  }
+  const { delta } = twofactor.verifyToken(secret, code, 10) || {};
+  return delta != null && delta >= 0;
+};
+
+Meteor.methods({
+  generate2faActivationQrCode(appName) {
+    const user = Meteor.user();
+
+    if (!user) {
+      throw new Meteor.Error(
+        400,
+        'There must be a user logged in to generate the QR code.'
+      );
+    }
+
+    const { username } = user;
+
+    const { secret, uri } = twofactor.generateSecret({
+      name: appName.trim(),
+      account: username,
+    });
+    const svg = new QRCode(uri).svg();
+
+    Meteor.users.update(
+      { username },
+      {
+        $set: {
+          'services.twoFactorAuthentication': {
+            secret,
+          },
+        },
+      }
+    );
+
+    return svg;
+  },
+  enableUser2fa(code) {
+    const user = Meteor.user();
+
+    if (!user) {
+      throw new Meteor.Error(400, 'No user logged in.');
+    }
+
+    const {
+      services: { twoFactorAuthentication },
+      username,
+    } = user;
+
+    if (!twoFactorAuthentication || !twoFactorAuthentication.secret) {
+      throw new Meteor.Error(
+        500,
+        'The user does not have a secret generated. You may have to call the function generateSvgCode first.'
+      );
+    }
+    if (!Accounts._isTokenValid(twoFactorAuthentication.secret, code)) {
+      throw new Meteor.Error(400, 'Invalid code.');
+    }
+
+    Meteor.users.update(
+      { username },
+      {
+        $set: {
+          'services.twoFactorAuthentication': {
+            ...twoFactorAuthentication,
+            type: 'otp',
+          },
+        },
+      }
+    );
+  },
+  disableUser2fa() {
+    const user = Meteor.user();
+
+    if (!user) {
+      throw new Meteor.Error(400, 'No user logged in.');
+    }
+
+    Meteor.users.update(
+      { username: user.username },
+      {
+        $unset: {
+          'services.twoFactorAuthentication': 1,
+        },
+      }
+    );
+  },
+  has2faEnabled(selector) {
```

Comment:
Getting the following error here:
```
I20220320-15:36:47.080(1)? Exception while invoking method 'has2faEnabled' Error: Did not check() all arguments during call to 'has2faEnabled'
I20220320-15:36:47.080(1)?     at ArgumentChecker.throwUnlessAllArgumentsHaveBeenChecked (packages/check/match.js:515:13)
```
Strange thing is that the error persist even if I add check for selector then it still throws. Possibly something weird going on with callback...

Reply:
Indeed it is. It is on the user settings page, so user needs to be logged in. I call in in React in a function that sets a state:
```ts
const check2FAEnabled = () => {
    Accounts.has2faEnabled(userId, (result: boolean) => {
      setIs2FAEnabled(result)
    })
  }
```

Reply:
I'll give it a try in about 10 hours after a good night sleep.

Reply:
Hey, I'm merging this yet as we could not reproduce the issue. 
We are generating the 2.7 version, and if that issue still happens we can fix it in 2.7, ok @StorytellerCZ?
Thanks!